RealmSwift/Impl/ComplexTypes.swift

```diff
@@ -301,8 +310,8 @@ extension RawRepresentable where Self: _OptionalPersistable, RawValue: _Optional
     }
 }
 
-extension RawRepresentable where Self: CaseIterable {
-    public static func _rlmDefaultValue() -> Self {
+extension PersistableEnum {
```

Comment:
I didn't see a reason why `extension RawRepresentable where Self: PersistableEnum` would be better than the non specialised equivalent.